code/core/src/components/components/Button/Button.tsx

```diff
@@ -12,13 +12,14 @@ import { InteractiveTooltipWrapper } from './helpers/InteractiveTooltipWrapper';
 import { useAriaDescription } from './helpers/useAriaDescription';
 
 export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  as?: 'a' | 'button' | 'div' | 'label' | typeof Slot;
   asChild?: boolean;
   size?: 'small' | 'medium';
   padding?: 'small' | 'medium' | 'none';
   variant?: 'outline' | 'solid' | 'ghost';
-  onClick?: (event: SyntheticEvent) => void;
   active?: boolean;
   disabled?: boolean;
+  readOnly?: boolean;
```

Comment:
I don't understand the purpose of `readOnly` here vs `disabled`.

If the goal is to allow using a button for decorative purposes (a bit like the Storybook window is built on the frontpage site), could we
* add a [presentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/presentation_role) role when the flag is set, and filter out any other aria attribute?
* name the flag more explicitly for its intended purpose so that people don't start using it in their UI code elsewhere?